src/langchain_google_firestore/vectorstores.py

```diff
@@ -0,0 +1,405 @@
+            distance_strategy: The distance strategy to use for calculating distances
+            between vectors. Defaults to DistanceStrategy.COSINE.
+
+        Raises:
+            ImportError: If the `firestore` package is not found.
+        """
+        try:
+            from google.cloud import firestore  # type: ignore
+        except ModuleNotFoundError as exc:
+            raise ImportError(IMPORT_ERROR_MSG) from exc
+
+        # Check if the client is provided, otherwise create a new client with
+        # the default client info.
+        self.client = client or firestore.Client()
+
+        client_agent = self.client._client_info.user_agent
+        if not client_agent:
+            self.client._client_info.user_agent = USER_AGENT
+        elif USER_AGENT not in client_agent:
+            self.client._client_info.user_agent = " ".join([client_agent, USER_AGENT])
+
+        if isinstance(collection, str):
+            self.collection = self.client.collection(collection)
+        else:
+            self.collection = collection
+
+        self.embedding = embedding
+        self.content_field = content_field
+        self.metadata_field = metadata_field
+        self.embedding_field = embedding_field
+        self.distance_strategy = distance_strategy
+        self.filters = filters
+
+    @property
+    def embeddings(self) -> Optional[Embeddings]:
+        return self.embedding
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Add or update texts in the vector store. If the `ids` are provided, and
+        a Firestore document with the same id exists, it will be updated.
+        Otherwise, a new Firestore document will be created.
+
+        Args:
+            texts: The texts to add to the vector store.
+            metadatas: The metadata to add to the vector store. Defaults to None.
+            ids: The document ids to use for the new documents. If not provided, new
+            document ids will be generated.
+
+        Returns:
+            List[str]: The list of document ids added to the vector store.
+        """
+
+        _ids = ids or []
+        db_batch = self.client.batch()
+
+        for batch in more_itertools.chunked(texts, WRITE_BATCH_SIZE):
+            texts_embs = self.embedding.embed_documents(batch)
+            for i, text in enumerate(batch):
+                doc_id = _ids[i] if i < len(_ids) else None
+                doc = self.collection.document(doc_id)
+                if doc_id is None:
+                    _ids.append(doc.id)
+                data = {
+                    self.content_field: text,
+                    self.embedding_field: Vector(texts_embs[i]),
+                }
+                if metadatas:
+                    data.update({self.metadata_field: metadatas[i]})
```

Comment:
`metadata` list is not enforced to be the same size as the `texts`

This can throw an error. Please add an assert at the beginning to avoid that.

nit: Also please change to
```
data[self.metadata_field] = metadatas[i]
```
For readibility